lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,58 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    collection.select { |x| x.is_a? (Numeric)}.count
   end
 
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
   def all_words_longer_than_length?(min_length)
+    collection.all? { |x| x.length > (min_length)}
   end
 
 
   # For the next 3 methods refer to: collect or map
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-any-3F
   def contains_a_word_longer_than?(min_length)
+    collection.any? { |x| x.length > (min_length)}
   end
 
   def capitalize_words
+    collection.map { |x| x.capitalize}
   end
 
   def square_numbers
+    collection.map { |x| x**2}
   end
 
   # Use symbol to proc notation on this one!
   def reverse_words
+    collection.map { |x| x.reverse}
   end
 
   # Refer to collect_concat or flat_map for these next two
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-flat_map
   # To get the order right for the tests, you'll need to use the absolute value method
   def positive_and_negative_numbers
+    collection.flat_map { |x| [x.abs, -x.abs]}
   end
 
   # Refer to detect or find
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-detect
   def find_first_awesome_person
+    collection.find { |x| x[:awesome] == true}
```

Comment:
In this case, `|person|` instead of `|x|`. And since `person[:awesome]` is already truthy, you don't need to test `== true`.